### PR TITLE
[fix]binlog load fails because txn exceeds the default value

### LINKED ISSUE #9468: [Bug] binlog load resume throws exception because txn exceeds max_running_txn_num_per_db
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

1.0

### What's Wrong?

2022-05-09 15:47:23,204 ERROR (Thread-95|192) [CanalSyncChannel.submit():258] encounter exception when submit in channel 94287, table: points_transaction_detail, batch: 81361
org.apache.doris.transaction.BeginTransactionException: errCode = 2, detailMessage = current running txns on db 81866 is 100, larger than limit 100
	at org.apache.doris.transaction.DatabaseTransactionMgr.checkRunningTxnExceedLimit(DatabaseTransactionMgr.java:1454) ~[palo-fe.jar:0.15-SNAPSHOT]
	at org.apache.doris.transaction.DatabaseTransactionMgr.beginTransaction(DatabaseTransactionMgr.java:300) ~[palo-fe.jar:0.15-SNAPSHOT]
	at org.apache.doris.transaction.GlobalTransactionMgr.beginTransaction(GlobalTransactionMgr.java:143) ~[palo-fe.jar:0.15-SNAPSHOT]
	at org.apache.doris.transaction.GlobalTransactionMgr.beginTransaction(GlobalTransactionMgr.java:109) ~[palo-fe.jar:0.15-SNAPSHOT]
	at org.apache.doris.load.sync.canal.CanalSyncChannel.beginTxn(CanalSyncChannel.java:129) ~[palo-fe.jar:0.15-SNAPSHOT]
	at org.apache.doris.load.sync.canal.CanalSyncChannel.submit(CanalSyncChannel.java:247) [palo-fe.jar:0.15-SNAPSHOT]
	at org.apache.doris.load.sync.canal.CanalSyncDataConsumer.executeOneBatch(CanalSyncDataConsumer.java:351) [palo-fe.jar:0.15-SNAPSHOT]
	at org.apache.doris.load.sync.canal.CanalSyncDataConsumer.process(CanalSyncDataConsumer.java:221) [palo-fe.jar:0.15-SNAPSHOT]
	at org.apache.doris.load.sync.SyncLifeCycle$2.run(SyncLifeCycle.java:48) [palo-fe.jar:0.15-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2022-05-09 15:47:23,292 ERROR (Thread-95|192) [CanalSyncDataConsumer.executeOneBatch():366] execute event has an error, data: header {
  version: 1
  logfileName: "bin.000034"
  logfileOffset: 831462819
  serverId: 1433116
  serverenCode: "UTF-8"
  executeTime: 1652068750000
  sourceType: MYSQL
  schemaName: "points_sharding_02"
  tableName: "points_transaction_detail"
  eventLength: 284
  eventType: INSERT
  props {
    key: "rowsCount"
    value: "1"
  }
}

### What You Expected?

If it exceeds the default value, prompt the user with a message

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hf200012**
# Proposed changes
binlog load Because txn exceeds the default value, resume is a failure, and a friendly prompt message is given to the user, instead of prompting success now, it still fails after a while, and the user will feel inexplicable

Issue Number: close #9468

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/load/sync/canal/CanalSyncChannel.java [hunk 1]
```diff
@@ -121,53 +123,60 @@ public void beginTxn(long batchId) throws UserException, TException, TimeoutExce
...
-                throw e;
-            }
-            try {
-                // async exec begin transaction
-                long txnId = txnExecutor.getTxnId();
-                if (txnId != -1L) {
-                    this.txnExecutor.beginTransaction(request);
-                    LOG.info("begin txn in channel {}, table: {}, label:{}, txn id: {}", id, targetTable, label, txnExecutor.getTxnId());
+            DatabaseTransactionMgr databaseTransactionMgr = globalTransactionMgr.getDatabaseTransactionMgr(db.getId());
+            if(databaseTransactionMgr.getRunningTxnNums() < Config.max_running_txn_num_per_db ) {
```
Comment (pengxiangyu) [inline fe/fe-core/src/main/java/org/apache/doris/load/sync/canal/CanalSyncChannel.java:127]: An unnecessary space after max_running_txn_num_per_db

File: fe/fe-core/src/main/java/org/apache/doris/load/sync/canal/CanalSyncChannel.java [hunk 2]
```diff
@@ -121,53 +123,60 @@ public void beginTxn(long batchId) throws UserException, TException, TimeoutExce
...
+                    LOG.warn ("Failed to begin txn in channel {}, table: {}, txn: {}, msg:{}", id, targetTable, txnExecutor.getTxnId (), e.getMessage ());
+                    throw e;
+                } catch ( TimeoutException | InterruptedException | ExecutionException e ) {
+                    LOG.warn ("Error occur while waiting begin txn response in channel {}, table: {}, txn: {}, msg:{}",
+                        id, targetTable, txnExecutor.getTxnId (), e.getMessage ());
+                    throw e;
+                }
+            } else {
+                String failMsg = "current running txns on db " + db.getId () + " is "
+                    + databaseTransactionMgr.getRunningTxnNums () + ", larger than limit " + Config.max_running_txn_num_per_db;
```
Comment (pengxiangyu) [inline fe/fe-core/src/main/java/org/apache/doris/load/sync/canal/CanalSyncChannel.java:178]: Add a LOG.warn() here is better.

**Review by pengxiangyu — DISMISSED** [review]
LGTM

**Review by xy720 — DISMISSED** [review]
LGTM

**Review by xy720 — APPROVED** [review]
LGTM

**Pull request merged**